torch/csrc/autograd/variable.h

```diff
@@ -507,18 +507,6 @@ inline Variable make_variable(
   return Variable();
 }
 
-// Tensor Conversion
-//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-// In the old days, these casts were checked, but now that every Tensor
-// is a Variable this cast is always valid
-inline Variable& as_variable_ref(at::Tensor& tensor) {
```

Comment:
Can we actually delete this? Do we have any nothing of BC-breaking changes to the cpp API / cpp extensions? cc @yf225

Reply:
It would probably be okay to remove `as_variable_ref` since this is mostly an internal API (we don't use it in any examples or tutorials), and also because the change on user end is easy (i.e. just remove the usage of `as_variable_ref` and everything will still work)